src/sorta.nim

```diff
@@ -523,6 +573,38 @@ when isMainModule:
     assert st.getOrDefault("www.dell.com") == "143.166.224.230"
     assert(st.entries == 16)
 
+    var keys = toSeq(st.keys())
+    for i in 0..<keys.len:
+      var key = "" & keys[i]
+      proc collect(): seq[string] = 
+        result = @[]
+        for position in st.scanFrom(key):
+          result.add(position.key)
+      echo "Checking key " & key
+      assert collect() == keys[i..<keys.len]
+      key[key.len-1].dec
+      echo "Checking key " & key
+      assert collect() == keys[i..<keys.len]
+      key = "" & keys[i]
+      key[key.len-1].inc
+      echo "Checking key " & key
+      assert collect() == keys[i+1..<keys.len]
+
+    for i in 0..<keys.len:
+      var from_key = "" & keys[i]
+      for j in i..<keys.len:
+        var to_key = "" & keys[j]
+        proc collect(): seq[string] =
```

Comment:
Can this proc be defined outside of the loops?

`proc collect(fromKey, toKey: string): seq[string]`